disnake/ext/commands/params.py

```diff
@@ -404,6 +508,9 @@ def __init__(
         min_length: Optional[int] = None,
         max_length: Optional[int] = None,
     ) -> None:
+        if autocomplete is not None:
+            classify_autocompleter(autocomplete)
```

Comment:
Pretty sure `classify_autocompleter` should also be called in `InvokableSlashCommand.__init__` and `SubCommand.__init__`, in a similar fashion (cc @EQUENOS) ?

Reply:
idts. If we were to put it there we allow autocompletes for commands to be outside of cogs, and I don't think that's a good idea right now.

Reply:
Hm, suppose this didn't actually change in those classes. In that case, the two new lines here should be removed?
Otherwise `commands.Param(autocomplete=SomeCog.stuff)` would work, which I don't think we'd want.

Reply:
https://github.com/DisnakeDev/disnake/pull/675/commits/526ed16ac62e4a36e987f3b29d0a36206e444fb2

Reply:
My original idea was to allow cross-cog autocompleters (with and without a cog arg), so shiftinv was right, I forgot to add classify_autocompleter calls in inits

Reply:
That's out of scope of this pull request. If we want that feature it should not be added as part of this pull. This pr is about injections, their typings, and autocompletes. ParamInfo is not injections, nor is it required for this pull. 

If that feature is desired (and I don't think it is worth adding, but we can talk about that in an issue) it should be added as a pull request, not as part of this pull.